networkx/algorithms/d_separation.py

```diff
@@ -11,6 +11,59 @@
 algorithm presented in [2]_.  Refer to [3]_, [4]_ for a couple of
 alternative algorithms.
 
+Here, we provide a brief overview of d-separation and related concepts that
+are relevant for understanding it:
+
+Blocking paths
+--------------
+
+Before we overview, we introduce the following terminology to describe paths:
+
+- "open" path: A path between two nodes that can be traversed
+- "blocked" path: A path between two nodes that cannot be traversed
+
+A **collider** is a triplet of nodes along a path that is like the following:
+``... u -> c <- v ...``), where 'c' is a common successor of 'u' and 'v'. A path
+through a collider is considered "blocked". When
+a node that is a collider, or a descendant of a collider is included in
+the d-separating set, then the path through that collider node is "open". If the
```

Comment:
Does the "descendent of a collider" have to be a collider along the path being considered? Or does the descendent have to be on the path? What is the relationship between the path and the collider and also the path and the descendant of the path?

I'm guessing that any colliders will block a path unless the collider node or any of its descendants are in the d-separation set.  So the d-separation set can include nodes very far from the collider and still "un-clock" paths through that collider if the collider has a directed path to a node in the d-separation set.

Reply:
Yeah any triple with `u -> w <- v` is unblocked if `w` or any of its descendants are in the d-separation set. If we then have `u -> w <- v -> .... -> a` and the rest of the path is not inside the d-separation set, then `u` and `a` are not d-separated on this path at least.

Reply:
That's an interesting nonlinear behavior for the blocking process.  The nodes in the d-separation set block paths -- except if they are in a particular graph structure. And then they open paths. 

Do you know the application that drove people to consider this idea of a "collider" and "opening the collider"?

Reply:
I think the reasons is that two node that collide at a third node should be "independent" of each other in the statistical sense. However, when you condition on the collider, you create a statistical dependency.

E.g. `x -> z <- y` would have `P(x,y) = P(x)P(y)`, but not `P(x, y | z) = P(x | z) P(y | z)`

Intuitively it means they are independent, but dependent when you condition on seeing the effect of both of them. E.g. water sprinklers and presence of clouds are independent, but if you condition on seeing the floor wet, then they are highly dependent.

Reply:
Does this mean that an "open" path means that probabilistic dependence occurs along that path?
And the d-separation set if the set of nodes you are wanting to condition your probabilities on?
Is the "d" of d-separation standing for dependence?

Reply:
Yep to both the first questions. 

Re "d" in d-separation, yes it is trying to tie the idea of "dependence" with separation in a graph. I think also, if interpreted probabilistically, d-separation is tied to conditional probabilistic independence assuming the Markov condition: https://en.wikipedia.org/wiki/Causal_Markov_condition. However, I guess in general d-separation doesn't have to do with anything probabilistically, but based on what I know, it's almost solely only used in probabilistic graphical models. 

^ If you want, I can add some details regarding the explicit connection with Markov condition of graphs and what it means probabilistically?

Reply:
Okay I've updated the doc string with a summary of the description I just added.